File: solutions/swift/secret-handshake/1/Sources/SecretHandshake/SecretHandshake.swift
func commands(number: Int) -> [String] {
  var result = [String]()
  if number & 0b00001 > 0 { result.append("wink") }
  if number & 0b00010 > 0 { result.append("double blink") }
  if number & 0b00100 > 0 { result.append("close your eyes") }
  if number & 0b01000 > 0 { result.append("jump") }
  if number & 0b10000 > 0 { result.reverse() }
  return result
}

File: solutions/zig/secret-handshake/2/secret_handshake.zig
const std = @import("std");
const mem = std.mem;

pub const Signal = enum { wink, double_blink, close_your_eyes, jump };

pub fn calculateHandshake(allocator: mem.Allocator, number: u5) mem.Allocator.Error![]const Signal {
    var result = std.ArrayList(Signal).init(allocator);
    defer result.deinit();
    if (number & 0b00001 > 0) try result.append(.wink);
    if (number & 0b00010 > 0) try result.append(.double_blink);
    if (number & 0b00100 > 0) try result.append(.close_your_eyes);
    if (number & 0b01000 > 0) try result.append(.jump);
    if (number & 0b10000 > 0) mem.reverse(Signal, result.items);
    return result.toOwnedSlice();
}
